receiver/prometheusexecreceiver/receiver_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheusexecreceiver
+
+import (
+	"context"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/config/configtest"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/consumer/pdatautil"
+	"go.opentelemetry.io/collector/exporter/exportertest"
+	"go.uber.org/zap"
+)
+
+// TestEndToEnd loads the test config.yaml and tests two things:
+// 1. makes sure the prometheus_exec config without an exec key throws an error
+// 2. An end-to-end test where metrics are scraped from a fake exporter that exposes Promtheus metrics,
+// and makes sure that exporter subprocess is restarted correctly
+func TestEndToEnd(t *testing.T) {
+	// Load the config from the yaml file
+	factories, err := componenttest.ExampleComponents()
+	assert.NoError(t, err)
+
+	factory := &Factory{}
+	factories.Receivers[factory.Type()] = factory
+
+	config, err := configtest.LoadConfigFile(t, path.Join(".", "testdata", "config.yaml"), factories)
+
+	assert.NoError(t, err)
+	assert.NotNil(t, config)
+
+	// Receiver without exec key, error expected and checked within function
+	assertErrorWhenExecKeyMissing(t, config.Receivers["prometheus_exec"])
+
+	// Normal test with port undefined by user
+	endToEndScrapeTest(t, config.Receivers["prometheus_exec/end_to_end_test/2"], "end-to-end port not defined")
```

Comment:
Not critical but `assertErrorWhenExecKeyMissing` seems very unrelated to `endToEndScrapeTest`. It would probably be better if these were two completely separate tests.